common/src/main/java/org/conscrypt/Hpke.java

```diff
@@ -0,0 +1,236 @@
+    } catch (Exception e) {
+      throw new IllegalStateException(
+          "Error while decrypting with the keys provided during setup recipient", e);
+    }
+  }
+
+  /**
+   * Hybrid Public Key Encryption (HPKE) encryption.
+   * <p>
+   * Returns a {@link HpkeResult} wrapper holding the resulting encapsulated key (enc) and
+   * ciphertext given a required plaintext and an optional associated data (aad). The API has a
+   * pre-requirement, an API call to set up[Mode]Sender is needed so the API could be properly
+   * initialized. Note: This API is stateful.
+   * <p><p>
+   * Multiple messages encryption:<p>
+   * If encrypting multiple messages that are expected to be decrypted in the same sequence as how
+   * they are encrypted, the API setup[Mode]Sender must be called just once before multiple calls
+   * to this API.
+   * <p><p>
+   * Single-shot encryption:<p>
+   * If encrypting a single message, the API setup[Mode]Sender must be called
+   * before every single message. Note: Calling setup[Mode][Sender|Recipient] resets the HPKE
+   * context.
+   *
+   * @param plaintext message that will be encrypted
+   * @param aad       optional associated data
+   * @return HpkeResult wrapping the resulting encapsulated key (enc) and ciphertext.
+   * @throws IllegalStateException if the API setup[Mode]Sender hasn't been called.
+   */
+  public HpkeResult seal(byte[] plaintext, byte[] aad) {
+    if (mSetupMode != Mode.SENDER) {
+      throw new IllegalStateException("Setup sender needs to be called before encryption");
+    }
+
+    Preconditions.checkNotNull(plaintext, "plaintext");
+    final byte[] ciphertext = NativeCrypto.EVP_HPKE_CTX_seal(mCtx, plaintext, aad);
+    return new HpkeResult(mEnc, ciphertext);
+  }
+
+  /**
+   * Hybrid Public Key Encryption (HPKE) secret exports.
+   * <p>
+   * Returns a {@link HpkeResult} wrapper holding the resulting encapsulated key (enc) and
+   * ciphertext given a required exporter desired output length and an optional exporter context.
+   * The API has a pre-requirement, an API call to setup[Mode][Sender|Recipient] is needed so the
+   * API could be properly initialized.
+   *
+   * @param length          expected output length
+   * @param exporterContext optional exporter context
+   * @return HpkeResult wrapping the resulting encapsulated key (enc) and the exported value.
+   * @throws IllegalArgumentException if the length is not valid based on the KDF specs.
+   * @throws IllegalStateException if the API setup[Mode][Sender|Recipient] hasn't been called.
+   */
+  public HpkeResult export(int length, byte[] exporterContext) {
+    if (mSetupMode == Mode.NONE) {
+      throw new IllegalStateException(
+          "Setup sender or recipient needs to be called before export");
+    }
+
+    mHpkeSuite.getKdf().validateExportLength(length);
+    final byte[] export = NativeCrypto.EVP_HPKE_CTX_export(mCtx, exporterContext, length);
+    return new HpkeResult(mEnc, export);
```

Comment:
Why is `export` returning `enc`? That doesn't make sense. Exporters return just the exported secret.
https://www.rfc-editor.org/rfc/rfc9180.html#section-5.3

Reply:
I have updated the code so setupBaseSender returns the enc instead and then export and seal can return just the given output. WDTY?